File: src/examples/mgl/samples/mouse.c
/****************************************************************************
*
*					SciTech Multi-platform Graphics Library
*
*  ========================================================================
*
*    The contents of this file are subject to the SciTech MGL Public
*    License Version 1.0 (the "License"); you may not use this file
*    except in compliance with the License. You may obtain a copy of
*    the License at http://www.scitechsoft.com/mgl-license.txt
*
*    Software distributed under the License is distributed on an
*    "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
*    implied. See the License for the specific language governing
*    rights and limitations under the License.
*
*    The Original Code is Copyright (C) 1991-1998 SciTech Software, Inc.
*
*    The Initial Developer of the Original Code is SciTech Software, Inc.
*    All Rights Reserved.
*
*  ========================================================================
*
* Language:		ANSI C
* Environment:	Any
*
* Description:  Simple demo showing how to use the MGL mouse functions.
*
****************************************************************************/

#include "mglsamp.h"

/*----------------------------- Global Variables --------------------------*/

char demoName[] = "mouse";

/*------------------------------ Implementation ---------------------------*/

ibool mouseCursorDemo(MGLDC *dc)
/****************************************************************************
*
* Function:     mouseCursorDemo
* Parameters:   dc  - Device context
*
* Description:  Display the mouse cursor on the screen, and change it to
*               a number of different styles.
*
****************************************************************************/
{
	cursor_t    *cursor;

	/* turn on the mouse */
	MS_show();

	statusLine("Press any key for Hour Glass cursor");
	waitEvent();

	if((cursor = MGL_loadCursor("wait.cur")) == NULL)
		MGL_fatalError("Unable to load WAIT.CUR cursor file");
	MS_setCursor(cursor);
	statusLine("Press any key for IBEAM cursor");
	waitEvent();

	MS_setCursor(MGL_DEF_CURSOR);
	MGL_unloadCursor(cursor);
	if((cursor = MGL_loadCursor("ibeam.cur")) == NULL)
		MGL_fatalError("Unable to load IBEAM.CUR cursor file");
	MS_setCursor(cursor);
	waitEvent();

	MS_setCursor(MGL_DEF_CURSOR);
	MGL_unloadCursor(cursor);

	if (!pause()) 
		return false;

	MS_hide();

	return true;
}

void demo(MGLDC *dc)
{
	/* Display title message at the top of the window. */
	mainWindow(dc,"Mouse Cursor Demonstration");

	/* set the background color to grey */
    if (MGL_getBitsPerPixel(dc) == 8) {
		MGL_setPaletteEntry(dc,1,100,100,100);
		MGL_realizePalette(dc,1,1,true);
		MGL_setBackColor(1);
		}
	else {
		MGL_setColorRGB(100,100,100);
		}

	MGL_clearViewport();

	if(!mouseCursorDemo(dc)) return;
}

File: src/examples/mgl/delphi/samples/mouse.pas
// Simple demo showing how to load a bitmap file from disk and
// display it on the screen.
unit Mouse;

interface

implementation

uses
    MGLSamp,
    MGLFx,
    SysUtils;

{-- MouseCursorDemo ----------------------------------------------------------}

// Display the mouse cursor on the screen, and change it to
// a number of different styles.

function    MouseCursorDemo(dc: PMGLDC): Boolean;
var
    cursor  : pcursor_t;
begin
    // First check to ensure that a mouse is actually installed in the
    // system. MS_init() will return 0 if no mouse is installed, or the
    // number of buttons if it is.
    if not MS_available then
        MGLError;

    // turn on the mouse
    MS_show;

    StatusLine('Press any key for Hour Glass cursor');
    WaitEvent;

    cursor := MGL_loadCursor('wait.cur');
    if cursor = nil then
        raise Exception.Create('Unable to load WAIT.CUR cursor file');
    try
        MS_setCursor(cursor);

        StatusLine('Press any key for IBEAM cursor');
        WaitEvent;

    finally
        MS_setCursor(MGL_DEF_CURSOR);
        MGL_unloadCursor(cursor);
    end;

    cursor := MGL_loadCursor('ibeam.cur');
    if cursor = nil then
        raise Exception.Create('Unable to load IBEAM.CUR cursor file');

    try
        MS_setCursor(cursor);
        WaitEvent;
    finally
        MS_setCursor(MGL_DEF_CURSOR);
        MGL_unloadCursor(cursor);
    end;

    if not Pause then
    begin
        Result := False;
        Exit;
    end;

    MS_hide;

    Result := True;
end;

{-- Demo ---------------------------------------------------------------------}

procedure   Demo(dc: PMGLDC);
begin
    // Display title message at the top of the window.
    MainWindow(dc,'Mouse Cursor Demonstration');

    MGL_setColorCI(MGL_DARKGRAY);
    MGL_setBackColor(MGL_getColor);
    MGL_clearViewport;
    MouseCursorDemo(dc);
end;

initialization
    RegisterDemo('Mouse','? Description should be corrected',Demo);
end.
